spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/BaseSparkAction.java

```diff
@@ -137,14 +139,22 @@ protected Table newStaticTable(TableMetadata metadata, FileIO io) {
     return new BaseTable(ops, metadataFileLocation);
   }
 
-  // builds a DF of delete and data file path and type by reading all manifests
   protected Dataset<FileInfo> contentFileDS(Table table) {
+    return contentFileDS(table, null);
+  }
+
+  protected Dataset<FileInfo> contentFileDS(Table table, Set<Long> snapshotIds) {
```

Comment:
If we decide to make any changes in 3.2 based on feedback, can we replicate them to 3.3 so that two versions are exactly the same?

Reply:
Yes, I tried to keep the two in sync, let me know if I missed something (might have missed some things after addressing Ryan's comments).